opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -92,28 +130,56 @@ public function index() {
+        $this->data['cancel'] = $this->url->link('extension/module', 'token=' . $this->session->data['token'], true);
         /**
          * Data for fields.
          */
         // Subdomain
         if (isset($this->request->post['smaily_for_opencart_subdomain'])) {
             $this->data['subdomain'] = $this->request->post['smaily_for_opencart_subdomain'];
         } else {
-            $this->data['subdomain'] = isset($settings['subdomain']) ? $settings['subdomain'] : '';
+            $this->data['subdomain'] = isset($settings['smaily_api_credentials']['subdomain']) ? $settings['smaily_api_credentials']['subdomain'] : '';
         }
         // Username
         if (isset($this->request->post['smaily_for_opencart_username'])) {
             $this->data['username'] = $this->request->post['smaily_for_opencart_username'];
         } else {
-            $this->data['username'] = isset($settings['username']) ? $settings['username'] : '';
+            $this->data['username'] = isset($settings['smaily_api_credentials']['username']) ? $settings['smaily_api_credentials']['username'] : '';
         }
         // Password
         if (isset($this->request->post['smaily_for_opencart_password'])) {
             $this->data['password'] = $this->request->post['smaily_for_opencart_password'];
         } else {
-            $this->data['password'] = isset($settings['password']) ? $settings['password'] : '';
+            $this->data['password'] = isset($settings['smaily_api_credentials']['password']) ? $settings['smaily_api_credentials']['password'] : '';
+        }
+
+        // Subscriber sync enabled.
+        if (isset($this->request->post['smaily_for_opencart_enable_subscribe'])) {
+            $this->data['subscribe_status'] = $this->request->post['smaily_for_opencart_enable_subscribe'];
+        } else {
+            $this->data['subscribe_status'] = isset($settings['smaily_customer_sync']['subscribe_status']) ? $settings['smaily_customer_sync']['subscribe_status'] : '';
+        }
+        // Customer sync additional fields.
+        if (isset($this->request->post['smaily_for_opencart_syncronize_additional'])) {
+            $this->data['subscribe_additional'] = $this->request->post['smaily_for_opencart_syncronize_additional'];
+        } else {
+            $this->data['subscribe_additional'] = isset($settings['smaily_customer_sync']['subscribe_additional']) ? $settings['smaily_customer_sync']['subscribe_additional'] : '';
         }
+        // Customer sync token.
+        if (! empty($this->request->post['smaily_for_opencart_sync_token'])) {
+            // Get sync token if user adds custom one.
+            $this->data['sync_token'] = $this->request->post['smaily_for_opencart_sync_token'];
+        } else {
+            if (! empty($this->config->get('smaily_for_opencart_sync_token'))) {
+                $this->data['sync_token'] = isset($settings['sync_token']) ? $settings['sync_token'] : '';
```

Comment:
There is no `sync_token` setting. Customer synchronization saves `subscribe_sync_token` under `smaily_customer_sync` key.